lib/srv/alpnproxy/auth/auth_proxy.go

```diff
@@ -113,13 +114,21 @@ func (s *AuthProxyDialerService) dialLocalAuthServer(ctx context.Context) (net.C
 	if len(authServers) == 0 {
 		return nil, trace.NotFound("empty auth servers list")
 	}
-	//TODO(smallinksy) Better support for HA. Add dial retry on auth network errors.
-	authServerIndex := rand.Intn(len(authServers))
-	conn, err := net.Dial("tcp", authServers[authServerIndex].GetAddr())
-	if err != nil {
-		return nil, trace.Wrap(err)
+	var errors []string
+
+	//select random server from list, try others if it's unavailable
+	for len(authServers) > 0 {
+		authServerIndex := rand.Intn(len(authServers))
+		var conn net.Conn
+		addr := authServers[authServerIndex].GetAddr()
+		conn, err = net.Dial("tcp", addr)
```

Comment:
While we're touching this:
```suggestion
		var d net.Dialer
		conn, err = d.DialContext(ctx, "tcp", addr)
```